pkg/relayer/session/claim.go

```diff
@@ -170,11 +162,20 @@ func (rs *relayerSessionsManager) waitForEarliestCreateClaimsHeight(
 	logger.Info().Msg("waiting & blocking until the earliest claim commit height for this supplier")
 
 	// Wait for the earliestSupplierClaimsCommitHeight to be reached before proceeding.
-	_ = rs.waitForBlock(ctx, earliestSupplierClaimsCommitHeight)
-
-	logger.Info().Msg("observed earliest claim commit height")
+	blockObserved := make(chan struct{}, 1)
+	go func() {
+		_ = rs.waitForBlock(ctx, earliestSupplierClaimsCommitHeight)
+		logger.Info().Msg("observed earliest claim commit height")
+		close(blockObserved)
+	}()
+
+	claimsFlushed, failedClaims := rs.createClaimRoots(sessionTrees)
+	if len(failedClaims) > 0 {
+		failedCreateClaimsSessionsCh <- failedClaims
```

Comment:
Can you log an error with the number of failed claim creations while waiting for the block?